pyproject.toml

```diff
@@ -47,3 +47,7 @@ Repository = "https://github.com/zeek/zeekscript"
 [tool.setuptools]
 packages = ["zeekscript"]
 script-files = ["zeek-format", "zeek-script"]
+
+[tool.ruff.lint]
+select = ["PL", "UP", "RUF", "N", "I"]
+ignore = ["PLR"]
```

Comment:
"PLR" seems to have a some useful rules. Is there too much noise ignoring them all?

https://docs.astral.sh/ruff/rules/#refactor-r

Reply:
I switched to ignoring only a subset.

- https://docs.astral.sh/ruff/rules/too-many-branches/
- https://docs.astral.sh/ruff/rules/too-many-statements/
- https://docs.astral.sh/ruff/rules/magic-value-comparison/

The first two are due to this code simply being in places too complicated, and I did not want to stack on a big, potentially risky refactor. The magic value instances flagged seemed valid uses which would not have benefited from explicit names (e.g., check for len==3 and then accessing element 0 and 2; I switched one instance of a duplicated magic value to use a variable `max_snippet_length`.